### ISSUE TITLE
CA1508: False positive for Expression<Func<>>

### ISSUE THREAD

**Issue by jnysteen** [dotnet/roslyn-analyzers#7342]
### Analyzer

**Diagnostic ID**: [CA1508](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/CA1508): `Avoid dead conditional code`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://learn.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 8.0.302](https://dotnet.microsoft.com/en-us/download/dotnet/8.0)

### Describe the bug

The analyzer warns about a variable `sortByExpression` being null, despite a non-null value being assigned to the variable in the line above.

### Steps To Reproduce

1. Run analyzers on the following code:
```csharp
public class Ca1508ReproItem
{
	public string Name { get; set; } = string.Empty;

	public static void MethodDoesntMatter()
	{
		Expression<Func<Ca1508ReproItem, string>> sortByExpression = x => x.Name;
		if (sortByExpression is null)
		{
			Console.WriteLine("wat");
		}
	}
}
```

### Expected behavior
No warning should be generated

### Actual behavior
The analyzer generates the warning `CA1508: 'sortByExpression is null' is always 'true'. Remove or refactor the condition(s) to avoid dead code` (the project in the screenshot treats warnings as errors, hence the red line)
![image](https://github.com/dotnet/roslyn-analyzers/assets/2863227/6a10de52-ce91-4c4c-8a14-4ee08940f15c)
